src/api/APIFactory.js

```diff
@@ -131,6 +143,14 @@ class APIFactory {
             this.fileAccessStatsAPI.destroy();
             delete this.fileAccessStatsAPI;
         }
+        if (this.tasksAPI) {
```

Comment:
This is getting pretty verbose/repeatable, any ideas on how we could refactor?

Reply:
I think the solution can be just aborting the request since axios allows us to do that.
https://github.com/axios/axios#cancellation

Before it wasn't possible, but now one can use that.

Reply:
Was just looking at that earlier. I can create another ticket for that, but I think its out of scope for this change

Reply:
can also just move the objects into another object.

this.apis = { file: fileAPI, ... }

in that way you can just iterate over the object.keys and destroy them all

Reply:
I added a ticket for this so we can get back to it later